soul-sync-data-center/soul-sync-data-nacos/src/main/java/org/dromara/soul/sync/data/nacos/handler/NacosCacheHandler.java

```diff
@@ -82,49 +85,79 @@ public NacosCacheHandler(final ConfigService configService, final PluginDataSubs
 
     protected void updatePluginMap(final String configInfo) {
         try {
-            List<PluginData> pluginDataList = GsonUtils.getInstance().fromList(configInfo, PluginData.class);
-            pluginDataList.forEach(pluginData -> Optional.ofNullable(pluginDataSubscriber).ifPresent(e -> e.unSubscribe(pluginData)));
-            pluginDataList.forEach(pluginData -> Optional.ofNullable(pluginDataSubscriber).ifPresent(e -> e.onSubscribe(pluginData)));
+            // Fix bug #656(https://github.com/dromara/soul/issues/656)
+            Map<String, Object> pluginDataMap = GsonUtils.getGson().fromJson(configInfo, Maps.newHashMap().getClass());
```

Comment:
refactor GsonUtils.getInstance().toObjectMap() ?

Reply:
> refactor GsonUtils.getInstance().toObjectMap() ?

ok, but this method is also used elsewhere, so I prefer to add a new one.